week2/game_2048/class_version/game_template.py

```diff
@@ -0,0 +1,236 @@
+"""
+Clone of 2048 game.
+"""
+import random
+
+from merge import merge
+
+#import poc_2048_gui
+
+# Directions, DO NOT MODIFY
+UP = 1
+DOWN = 2
+LEFT = 3
+RIGHT = 4
+
+# Offsets for computing tile indices in each direction.
+# DO NOT MODIFY this dictionary.
+OFFSETS = {UP: (1, 0),
+           DOWN: (-1, 0),
+           LEFT: (0, 1),
+           RIGHT: (0, -1)}
+
+class TwentyFortyEight:
+    """
+    Class to run the game logic.
+    """
+
+    def __init__(self, grid_height=4, grid_width=4):
+        self._grid_height = grid_height
+        self._grid_width = grid_width
+        self._grid = self.generate_empty_grid()
+        self._move_grid_indices = self.generate_move_grid_indices()
+
+    def set_move_grid_indices(self, move_dict):
+        """
+        Set the value of move_grid_indices
+        """
+        self._move_grid_indices = move_dict
+
+    def get_move_grid_indices(self):
```

Comment:
Python offers you setters and getters using decorators. You should prefer those.